### PR TITLE
assign_value operator

### PR DESCRIPTION
**Pull request by emailweixu**
We need this operator to assign value to a tensor and the values are stored in the program so that they can be used independent of python.

### CODE REVIEW

File: paddle/operators/assign_value_op.cc [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+  void InferShape(framework::InferShapeContext *ctx) const override {
+    PADDLE_ENFORCE(ctx->HasOutput("Out"),
+                   "Output(Out) of AssignValueOp should not be null.");
+    auto shape = ctx->Attrs().Get<std::vector<int>>("shape");
+    ctx->SetOutputDim("Out", framework::make_ddim(shape));
+  }
+
+ protected:
+  framework::OpKernelType GetActualKernelType(
```
Comment (JiayiFeng) [inline paddle/operators/assign_value_op.cc]: The function name should be `GetExpectedKernelType`.

See doc: https://github.com/PaddlePaddle/Paddle/blob/develop/doc/design/kernel_hint_design.md#final-choice

File: paddle/operators/assign_value_op.cc [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+    AddOutput("Out", "(Tensor) Output tensor of assign_value operator.");
+    AddAttr<std::vector<int>>("shape",
+                              "(vector<int>) "
+                              "Shape of values.");
+    AddAttr<int>("dtype", "data type of values")
+        .InEnum({framework::proto::DataType::INT32,
+                 framework::proto::DataType::FP32});
+    AddAttr<std::vector<float>>("fp32_values", "store the float values")
+        .SetDefault({});
+    AddAttr<std::vector<int>>("int32_values", "store the int values")
```
Comment (JiayiFeng) [inline paddle/operators/assign_value_op.cc]: Op's attribute will be stored into the protobuf. However, the size of a protobuf object is limited(64M). I'm afraid there might be some problem if the tensor to be assigned is huge.

File: paddle/operators/assign_value_op.cc (hunk 1 above)
Comment (emailweixu) [reply paddle/operators/assign_value_op.cc]: Fixed.
The interface changed before I submitted the PR.

File: paddle/operators/assign_value_op.cc (hunk 2 above)
Comment (emailweixu) [reply paddle/operators/assign_value_op.cc]: Yes. If the arry is too big, we may have to use a separate file. But if many cases where only a small matrix is used, it's quite inconvenient to use a separate file to array the matrix.

File: paddle/operators/assign_value_op.h [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+        break;
+      case framework::proto::DataType::FP32:
+        value_name = "fp32_values";
+        break;
+      default:
+        PADDLE_THROW("Unsupported dtype for assign_value_op: %d", dtype);
+        break;
+    }
+    auto values = ctx.Attr<std::vector<T>>(value_name);
+    Copy(dst, values.data(), sizeof(T) * values.size(), ctx);
```
Comment (dzhwinter) [inline paddle/operators/assign_value_op.h]: We have a uniform Copy interface of  https://github.com/PaddlePaddle/Paddle/blob/develop/paddle/framework/tensor_util.h#L101

Comment (emailweixu) [reply paddle/operators/assign_value_op.h]: Fixed.

**Pull request merged**